bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/CommonInterface.java

```diff
@@ -0,0 +1,194 @@
+
+    void triggerChannel(String channelID, String event);
+
+    void updateThing(Thing thing);
+
+    @Nullable
+    Bridge getBridge();
+
+    default @Nullable CommonInterface getBridgeHandler() {
+        Bridge bridge = getBridge();
+        return bridge != null && bridge.getHandler() instanceof AccountHandler ? (AccountHandler) bridge.getHandler()
+                : null;
+    }
+
+    default @Nullable ApiBridgeHandler getRootBridge() {
+        Bridge bridge = getBridge();
+        BridgeHandler bridgeHandler = null;
+        if (bridge != null) {
+            bridgeHandler = bridge.getHandler();
+            while (bridgeHandler != null && !(bridgeHandler instanceof ApiBridgeHandler)) {
+                bridge = ((CommonInterface) bridgeHandler).getBridge();
+                bridgeHandler = bridge != null ? bridge.getHandler() : null;
+            }
+        }
+        return (ApiBridgeHandler) bridgeHandler;
+    }
+
+    default Optional<NetatmoServlet> getServlet() {
+        ThingHandler handler = getThing().getHandler();
+        Bridge root = null;
+        if (handler instanceof NAThingHandler) {
+            CommonInterface bridgeHandler = ((NAThingHandler) handler).getBridgeHandler();
+            if (bridgeHandler != null) {
+                root = bridgeHandler.getBridge();
+            }
+        } else if (handler instanceof AccountHandler) {
+            root = ((AccountHandler) handler).getBridge();
+        }
+        if (root instanceof ApiBridgeHandler) {
+            return ((ApiBridgeHandler) root).getServlet();
+        }
+        return Optional.empty();
+    }
+
+    default @Nullable String getBridgeId() {
+        CommonInterface bridge = getBridgeHandler();
+        return bridge != null ? bridge.getId() : null;
+    }
+
+    default void expireData() {
+        getCapabilities().values().forEach(cap -> cap.expireData());
+    }
+
+    default String getId() {
+        return (String) getThing().getConfiguration().get("id");
+    }
+
+    default Stream<Channel> getActiveChannels() {
+        return getThing().getChannels().stream()
+                .filter(channel -> ChannelKind.STATE.equals(channel.getKind()) && isLinked(channel.getUID()));
+    }
+
+    default Optional<CommonInterface> getHomeHandler() {
+        CommonInterface bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            return bridgeHandler.getCapabilities().get(HomeCapability.class).isPresent() ? Optional.of(bridgeHandler)
+                    : Optional.empty();
+        }
+        return Optional.empty();
+    }
+
+    default Stream<CommonInterface> getActiveChildren() {
+        Thing thing = getThing();
+        if (thing instanceof Bridge) {
+            return ((Bridge) thing).getThings().stream().filter(
+                    t -> t.getStatus().equals(ThingStatus.ONLINE) || t.getStatus().equals(ThingStatus.INITIALIZING))
```

Comment:
It's needed, or else INITIALIZING things will never come ONLINE. In fact the idea is more to take apart OFFLINE, REMOVING...modules.

Reply:
INITIALIZING was probably required because you forgot to set an initial state (UNKNOWN) in initialize().
Now that each thing is initialized with UNKNOWN, I don't think it is still necessary to consider INITIALIZING. By the way, if the thing is in INITIALIZING state, that would mean that initialize() is not yet called and so it makes sense to not consider yet.

Reply:
Correct, and changed accordingly.